src/main/java/com/devonfw/ide/sonarqube/common/impl/check/DevonArchitectureCheck.java

```diff
@@ -124,47 +124,56 @@ public void visitClass(ClassTree classTree) {
 
   private void checkIfDisallowed(String className, Tree tree) {
 
-    if (tree.is(Tree.Kind.INFERED_TYPE)) {
-      return;
-    }
-    if ((this.sourcePackage == null) || !this.sourcePackage.isValid()) {
-      return;
-    }
     int lastDot = className.lastIndexOf('.');
-    if (lastDot <= 0) {
-      return;
-    }

```

Comment:
I didn't remove them directly, but put them in new private methods. The problem here was that the method checkIfDisallowed() was too complex in terms of if-conditions. So I sourced some of the conditions into seperate methods to reduce the complexity. 
In line 134 there is a check that verifies if lastDot is less or equal to zero. If it is, the method returns.

Reply:
I think you are not getting what I meant.
in case `tree.is(Tree.Kind.INFERED_TYPE)` is `true` we have an inferred type and then `className` may be `null`. This will cause a NPE in Line 133.
You moved that code to a method to make SonarQube happy what is good. But you also changed the order when the checks take place and the method returns.
IMHO this is still a bug in your PR.

Reply:
IMHO possible from this case (I slightly remember I once hit such a strange case of SonarQube whilst debugging):
`String superClassTypeName = superClass.symbolType().fullyQualifiedName();`

However, to make it more explicit you can just add this on top:
```
if (className == null) {
  return;
}
```
Then this should be ready to be merged. I assume that if `lastDot` is positive we can also not hit the case that `sourcePackage` is `null` as this condition is also gone after your refctoring.

Reply:
Or you just move this block back to the top of the method:
```
    if (!isTreeAndSourcePackageValid(tree)) {
      return;
    }
```

Reply:
The method now starts with: 

` if (!isTreeAndSourcePackageValid(tree) || className == null) {
    return;
} `

Is this how you meant it?